Fixes proposed for PR #82

SeanFeldman opened the pull request:
TL&DR: yes, read and checked.

Improvements suggested for PR #82 that addresses issue #33 

@nemakam I might missed a few things, sorry 😊

msftclas commented:
Hi __@SeanFeldman__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

nemakam commented on test/Microsoft.Azure.ServiceBus.UnitTests/QueueClientTests.cs:
```diff
@@ -174,7 +174,7 @@ public sealed class QueueClientTests : SenderReceiverClientTestBase
             var queueClient = (QueueClient)messagingFactory.CreateQueueClientFromConnectionString(TestUtility.GetEntityConnectionString(queueName), ReceiveMode.ReceiveAndDelete);
```
(QueueClient) cast can be removed for all the tests and the object can be directly passed everywhere instead of InnerReceiver or InnerSender

nemakam commented on test/Microsoft.Azure.ServiceBus.UnitTests/QueueClientTests.cs:
```diff
@@ -174,7 +174,7 @@ public sealed class QueueClientTests : SenderReceiverClientTestBase
             try
             {
-                await this.ReceiveShouldReturnNoLaterThanServerWaitTimeTestCase(queueClient.InnerSender, queueClient.InnerReceiver, messageCount);
+                await this.ReceiveShouldReturnNoLaterThanServerWaitTimeTestCase(queueClient.InnerReceiver);
```
Based on previous comment, remove the 'InnerReceiver' .
Same for all other occurrences

nemakam commented on test/Microsoft.Azure.ServiceBus.UnitTests/SenderReceiverClientTestBase.cs:
```diff
@@ -124,7 +124,7 @@ public abstract class SenderReceiverClientTestBase
             await Task.Delay(TimeSpan.FromSeconds(10));
 
-            DateTime lockedUntilUtcTime = await messageReceiver.RenewLockAsync(receivedMessages.First().LockToken);
+            var lockedUntilUtcTime = await messageReceiver.RenewLockAsync(receivedMessages.First().LockToken);
```
I don't think `var` should ALWAYS be used. For me, I prefer `var` wherever the type is understood and obvious when you don't have prior knowledge of the whole system. `var size = 1` is obvious. But in this case unless you open this in a IDE, you cant be sure whether the object returned is DateTime / DateTimeOffset or any other custom implementation.

SeanFeldman replied:
@nemakam Agree. With `lockedUntilUtcTime` I would argue you're dealing with a date time that cares for UTC and not an offset. If you need to emphasize the offset, you'll do so with a variable name. For example: `localTimeWithOffset` or `localTimeWithTimezone` would indicate that you're looking at a time with offset component. 

Now, for the argument
> But in this case unless you open this in a IDE, you cant be sure whether the object returned is DateTime / DateTimeOffset or any other custom implementation.

The codebase is no longer a few files. To actually read it and understand, a review of a diff on GH is not enough. It either requires to pull the code down and see it all, or requires to be familiar with the code. I'd say if you pull the code down, you will be using an IDE.

We need to understand how vehicles operate to drive those, but do we really need to know to the low level how things work exactly? You know there's an engine, not Engive v6 3.5L blah blah blah. I see `var` as engine. Specifications (if needed) can be a part of the variable name.

Hopefully I have managed to convey my POV 😃

nemakam replied:
@SeanFeldman, this particular instance might fall in the grey area because of the naming convention of the variable. And I guess this is one of those never ending coding practices debate. Personally, I'd prefer that from next set of PRs, we use `var` with discretion and not everytime ReSharper suggests it :) 

Regarding
> The codebase is no longer a few files. To actually read it and understand, a review of a diff on GH is not enough. It either requires to pull the code down and see it all, or requires to be familiar with the code. I'd say if you pull the code down, you will be using an IDE.

I neither pull down the code and do the review in IDE, nor do I claim to be an expect with the entire code...

And regarding
> I see `var` as engine

I'm just looking out for people who see it as `object` and not engine.

SeanFeldman replied:
Fair point. Agree to take it case by case.
R# can be harmful sometimes. That's why it's just a tool, not a developer replacement 😆

SeanFeldman commented:
@nemakam I've rebased my branch on top of `upstream/exposure` so that we don't work on two different PRs? Changed GH to target `upstream/dev` with this PR. I hope that's not an issue.

nemakam commented:
@SeanFeldman, Sure. Not an issue. I wonder why take the trouble of this when its going to get squashed anyway.

SeanFeldman commented:
An "opsie" moment? I realized that it was a bit too early after I've pushed...
Since we'll target with this PR `dev` anyways, no harm is done I thought. Sorry for that 😊

SeanFeldman commented:
Fixed the build. 💚

jtaubensee commented:
Hi @SeanFeldman - For simplicity's sake, lets close this PR, and reopen once #82 is merged. Does that work for you? I think it will make the history a little bit more clear.

SeanFeldman commented:
👍 or just cherry pick those changes 😉
